src/app/content/components/Page/highlightManager.ts

```diff
@@ -49,11 +55,16 @@ const onSelectHighlight = (
   services: Services,
   highlights: Highlight[],
   highlight: Highlight | undefined
-) => defer(() => {
-  if (highlights.length > 0 || !highlight || services.getProp().focused) {
+) => defer(async() => {
+  if (highlights.length > 0 || !highlight) {
     return;
   }
 
+  if (services.getProp().focused && services.getProp().hasUnsavedHighlight) {
+    const didConfirm = await showConfirmation();
+    return didConfirm && services.getProp().clearFocus();
```

Comment:
don't you want to return if they _don't_ confirm, and just let it do the stuff below if they do?